lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -47,15 +53,32 @@ public function getAttribute( $userId, $attributeName, $default = null ) {
 	}
 
 	private function loadAttributes( $userId ) {
-		if ( !isset( $this->attributes[$userId] ) ) {
-			$this->attributes[$userId] = [];
+
+		if ( isset( $this->attributes[$userId] ) ) {
+			return $this->attributes[$userId];
+		}
+
+		$attributes = $this->loadFromMemcache( $userId );
```

Comment:
I didn't think about that. I borrowed this approach from what the [preferences guys are doing](https://github.com/Wikia/app/blob/dev/lib/Wikia/src/Service/User/Preferences/PreferenceServiceImpl.php). For a second pass we could consider updating though.